### PR TITLE
store: Don't trip over old errors with invalid block hash/range combinations

### PR DESCRIPTION
**Pull request by lutter**

### CODE REVIEW

**Review by leoyvens — CHANGES_REQUESTED** [review]
We should also be able to deal with the block hash being null, as far as I know it's very much possible for a fatal error to not be associated with a block.

**Comment by lutter** [comment]
Take 2

File: store/postgres/src/detail.rs [hunk 1]
```diff
@@ -140,8 +140,14 @@ impl TryFrom<ErrorDetail> for SubgraphError {
             block_range,
         } = value;
         let block_number = crate::block_range::first_block_in_range(&block_range).map(|n| n.into());
-        let block_ptr = block(&subgraph_id, "fatal_error", block_hash, block_number)?
-            .map(|block| block.to_ptr());
+        // In existing databases, we have errors that have a `block_range` of
+        // `UNVERSIONED_RANGE`, which leads to `None` as the block number, but
+        // has a hash. In that case, just return a block pointer of `None`
+        let block_ptr = match (&block_number, &block_hash) {
+            (Some(_), Some(_)) => block(&subgraph_id, "fatal_error", block_hash, block_number)?
```
Comment (leoyvens) [inline store/postgres/src/detail.rs:147]: Going through `block` function is complicating this code, we should not convert to `BigDecimal` and construct the `EthereumBlock` directly in this branch.

**Review by leoyvens — CHANGES_REQUESTED** [review]

**Review by fordN — APPROVED** [review]
LGTM 👍

**Comment by lutter** [comment]
Take 3

**Review by leoyvens — APPROVED** [review]

**Pull request merged**